alb.tf

```diff
@@ -78,13 +78,75 @@ resource "aws_alb_listener" "http" {
 }
 
 # We may want to create this resource without the loop if the path_patterns ever break the pattern of being the name of the service
-resource "aws_alb_listener_rule" "this" {
+resource "aws_alb_listener_rule" "http" {
   for_each = {
     for key, value in aws_alb_target_group.this : key => value
     if local.service_data[key].public == true
   }
   listener_arn = aws_alb_listener.http.arn
 
+  dynamic "action" {
+    for_each = var.certificate_arn != "" ? [] : [each.value]
+    content {
+      type             = "forward"
+      target_group_arn = each.value.arn
+      # terraform will complain that we have a redirect action and a forward action but the issue disappears on a subsequent apply
```

Comment:
Can you elaborate on this terraform complaint?

Reply:
It's a deprecation warning saying we shouldn't have a redirect action, not anything that throws.

This happens when I create the infra with a cert and then run it without the cert (destroying all the ssl/443/cert related infra). That shouldn't be a typical case. 

The resource doesn't explicitly delete the redirect part of the action, but it does implicitly because AWS doesn't allow it, which is why we don't see it on the follow-up run.